src/garage/tf/q_functions/continuous_cnn_q_function.py

```diff
@@ -0,0 +1,259 @@
+                    type(self).__name__,
+                    type(env_spec.observation_space).__name__,
+                    env_spec.observation_space.shape))
+
+        super().__init__(name)
+        self._env_spec = env_spec
+        self._filter_dims = filter_dims
+        self._num_filters = num_filters
+        self._strides = strides
+        self._hidden_sizes = hidden_sizes
+        self._action_merge_layer = action_merge_layer
+        self._padding = padding
+        self._max_pooling = max_pooling
+        self._pool_strides = pool_strides
+        self._pool_shapes = pool_shapes
+        self._cnn_hidden_nonlinearity = cnn_hidden_nonlinearity
+        self._hidden_nonlinearity = hidden_nonlinearity
+        self._hidden_w_init = hidden_w_init
+        self._hidden_b_init = hidden_b_init
+        self._output_nonlinearity = output_nonlinearity
+        self._output_w_init = output_w_init
+        self._output_b_init = output_b_init
+        self._layer_normalization = layer_normalization
+
+        self._obs_dim = self._env_spec.observation_space.shape
+        self._action_dim = self._env_spec.action_space.shape
+
+        self.model = CNNMLPMergeModel(
+            filter_dims=self._filter_dims,
+            num_filters=self._num_filters,
+            strides=self._strides,
+            hidden_sizes=self._hidden_sizes,
+            action_merge_layer=self._action_merge_layer,
+            padding=self._padding,
+            max_pooling=self._max_pooling,
+            pool_strides=self._pool_strides,
+            pool_shapes=self._pool_shapes,
+            cnn_hidden_nonlinearity=self._cnn_hidden_nonlinearity,
+            hidden_nonlinearity=self._hidden_nonlinearity,
+            hidden_w_init=self._hidden_w_init,
+            hidden_b_init=self._hidden_b_init,
+            output_nonlinearity=self._output_nonlinearity,
+            output_w_init=self._output_w_init,
+            output_b_init=self._output_b_init,
+            layer_normalization=self._layer_normalization)
+
+        self._initialize()
+
+    def _initialize(self):
+        obs_ph = tf.compat.v1.placeholder(tf.float32, (None, ) + self._obs_dim,
+                                          name='state')
+        action_ph = tf.compat.v1.placeholder(tf.float32,
+                                             (None, ) + self._action_dim,
+                                             name='action')
+        if isinstance(self._env_spec.observation_space, akro.Image):
+            obs_ph = obs_ph / 255.0
+
+        with tf.compat.v1.variable_scope(self.name) as vs:
+            self._variable_scope = vs
+            self.model.build(obs_ph, action_ph)
+
+        self._f_qval = tf.compat.v1.get_default_session().make_callable(
+            self.model.networks['default'].outputs,
+            feed_list=[obs_ph, action_ph])
+
+    @property
+    def inputs(self):
+        """Return the input tensor.
+
+        Returns:
+            tf.Tensor: The input tensors of the model.
```

Comment:
properties should be documented as attributes, see https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html#example-google for an example

it seems this has multiple return values? can you tell us nothing more than "input tensors"?

Reply:
fixed